### PR TITLE
[confmap] Mark module as stable

### LINKED ISSUE #9379: Stabilize module confmap
Before stabilizing a module, an approver or maintainer must make sure that the following criteria have been met for at least two successive minor version releases:

- [x] No open issues or PRs in the module that would require breaking changes
- [x] No TODOs in the module code that would require breaking changes
- [x] No deprecated symbols in the module
- [x] No symbols marked as experimental in the module
- [x] The module follows the [Coding guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md)

Please also make sure to publicly announce our intent to stabilize the module on:

- [x] The #otel-collector CNCF Slack Channel
- [x] The #opentelemetry CNCF Slack channel
- [x] A Collector SIG meeting (if unable to attend, just add to the agenda)

To help other people verify the above criteria, please link to the announcement and other links used to complete the above in a comment on this issue.

Once all criteria are met, close this issue by moving this module to the `stable` module set.

### PR DESCRIPTION
**Pull request by mx-psi**
Fixes #9379.

I removed two TODOs because I think we can just stay with the current behavior, but would want confirmation from @codeboten about this since it's mentioned here https://github.com/open-telemetry/opentelemetry-collector/pull/4880#pullrequestreview-886201749

### CODE REVIEW

File: confmap/confmap_test.go [hunk 1]
```diff
@@ -113,7 +113,6 @@ func TestExpandNilStructPointersHookFunc(t *testing.T) {
 	// assert.False(t, *cfg.Boolean)
 	assert.Nil(t, cfg.Struct)
 	assert.NotNil(t, cfg.MapStruct)
-	// TODO: Investigate this unexpected result.
```
Comment (mx-psi) [inline confmap/confmap_test.go:116]: This and the other TODO were added in #4880. I don't understand why this is unexpected but in any case I feel like we should just live with it, I don't see a benefit in changing this right now

**Review by songy23 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]
🥳

**Review by codeboten — APPROVED** [review]
I agree that those TODOs are probably not going to get addressed in the foreseeable future and they haven't caused issues thus far.

File: confmap/confmap_test.go (hunk 1 above)
Comment (codeboten) [reply confmap/confmap_test.go:116]: agreed

**Review by atoulme — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]
LGTM 👍

**Comment by TylerHelmuth** [comment]
@mx-psi before you merge take a look at https://github.com/open-telemetry/opentelemetry-collector/pull/10950. I believe the issue is the result of a regression (when I tested the scenario with 0.90.0 the collector did not panic).

**Pull request merged**